app/redux/shared_actions/SetsActionCreators.js

```diff
@@ -66,11 +68,53 @@ export const saveHistorySet = (setID, exercise = null, weight = null, metric = n
     return action;
 };
 
-export const endSet = () => (dispatch, getState) => {
+export const endSet = (was_sanity_check=false, duration=0, manually_started=false) => (dispatch, getState) => {
     var state = getState();
     var workoutData = state.sets.workoutData;
     var set = workoutData[workoutData.length - 1];
+    var prevSet = workoutData[workoutData.length - 2];
     var defaultMetric = state.settings.defaultMetric;
+    var num_fields_entered = 0;
+    var is_previous_set_fields_filled = null;
```

Comment:
Don't need to declare it here, just use vars in the if statement below

Reply:
Done